### PR TITLE
fximporter: support newer torch versions

### PR DESCRIPTION
**Pull request by dan-garvey**
uses version checking since attributes exist in both versions, the only thing that changes is what we're receiving as an fx graph

### CODE REVIEW

File: python/torch_mlir/extras/fx_importer.py [hunk 1]
```diff
@@ -220,19 +220,35 @@
...
-}
-
-SYMBOLIC_OP_TO_TORCH_OP = {
-    (torch.ops.aten.sym_size, 1): torch.ops.aten.size.default,
-    (torch.ops.aten.sym_size, 2): torch.ops.aten.size.int,
-    (torch.ops.aten.sym_stride, 1): torch.ops.aten.stride.default,
-    (torch.ops.aten.sym_stride, 2): torch.ops.aten.stride.int,
-    (torch.ops.aten.sym_numel, 1): torch.ops.aten.numel.default,
-}
+# this allows stripping of cuda versions like: "+cu113"
```
Comment (stellaraccident) [inline python/torch_mlir/extras/fx_importer.py:223]: I don't understand what this comment means?

**Comment by dan-garvey** [comment]
@stellaraccident I know you had some thoughts on how the tests for fx_importer should be unittestier for torch_mlir. I'm happy to pick that up if you want to give me a pointer to what you'd like changed.

File: python/torch_mlir/extras/fx_importer.py [hunk 2]
```diff
@@ -220,19 +220,35 @@
...
+if torch.__version__.split("+")[0] <= "2.1.0":
```
Comment (stellaraccident) [inline python/torch_mlir/extras/fx_importer.py:224]: Why can't we just do a direct comparison (what is the split doing?)

File: python/torch_mlir/extras/fx_importer.py [hunk 3]
```diff
@@ -220,19 +220,35 @@
...
+    SYMBOLIC_TORCH_OPS = {
+        torch.ops.aten.sym_size,
+        torch.ops.aten.sym_stride,
+        torch.ops.aten.sym_numel,
+    }
+
+    SYMBOLIC_OP_TO_TORCH_OP = {
```
Comment (stellaraccident) [inline python/torch_mlir/extras/fx_importer.py:231]: I've thought for quite a while that this whole fragment is hinting at a design flaw, and now we're adding to it. Can you please precede this section with a comment about what these maps are and what their fields mean?

File: python/torch_mlir/extras/fx_importer.py [hunk 4]
```diff
@@ -1212,7 +1230,10 @@ def _import_symbolic_torch_op(
             ), f"Unsupported builtin function for symbolic types: {target} with args {node.args}"
             concrete_target = getattr(torch_op, op_overload)
         else:
-            concrete_target = SYMBOLIC_OP_TO_TORCH_OP.get((target, len(node.args)))
+            if torch.__version__.split("+")[0] <= "2.1.0":
```
Comment (stellaraccident) [inline python/torch_mlir/extras/fx_importer.py:1233]: I think we should have these version checks at the module level bound to constants like `_IS_TORCH_2_1_OR_EARLIER`. saves some inline work and also makes the code more readable and clear for when we can drop branches.

**Review by stellaraccident — CHANGES_REQUESTED** [review]
Thanks, a few changes I'd like to see as this part was already limping and this adds to it.

File: python/torch_mlir/extras/fx_importer.py (hunk 1 above)
Comment (dan-garvey) [reply python/torch_mlir/extras/fx_importer.py:223]: when torch installs with cuda,  torch.__version__ = "2.1.0+cu113". In order to support both "2.1.0" and "2.1.0+cu*" split at the plus and then compare. 

"2.1.0+cu113" <= "2.1.0" resolves to False. Maybe there is a fancier way to do it but 🤷

File: python/torch_mlir/extras/fx_importer.py [hunk 5]
```diff
@@ -220,19 +220,45 @@
...
+# set and just check key existence in SYMBOLIC_OP_TO_TORCH_OP
+
+if _IS_TORCH_2_1_OR_EARLIER:
+    SYMBOLIC_TORCH_OPS = {
+        torch.ops.aten.sym_size,
+        torch.ops.aten.sym_stride,
+        torch.ops.aten.sym_numel,
+    }
+
+    SYMBOLIC_OP_TO_TORCH_OP = {
```
Comment (stellaraccident) [inline python/torch_mlir/extras/fx_importer.py:241]: One more nit: Please say what the number in the tuple key means.

**Review by stellaraccident — APPROVED** [review]
One nit then lgtm. Thanks for the comments and ergonomics.

**Pull request merged**